tests/components/script/test_init.py

```diff
@@ -213,31 +194,60 @@ def record_call(service):
         assert calls[1].context is context
         assert calls[1].data["hello"] == "universe"
 
-    def test_reload_service(self):
-        """Verify that the turn_on service."""
-        assert setup_component(
-            self.hass,
-            "script",
-            {"script": {"test": {"sequence": [{"delay": {"seconds": 5}}]}}},
-        )
 
-        assert self.hass.states.get(ENTITY_ID) is not None
-        assert self.hass.services.has_service(script.DOMAIN, "test")
+invalid_configs = [
+    {"test": {}},
+    {"test hello world": {"sequence": [{"event": "bla"}]}},
+    {"test": {"sequence": {"event": "test_event", "service": "homeassistant.turn_on"}}},
+    {"test": {"sequence": [], "mode": "parallel", "queue_size": 5}},
+]
 
-        with patch(
-            "homeassistant.config.load_yaml_config_file",
-            return_value={
-                "script": {"test2": {"sequence": [{"delay": {"seconds": 5}}]}}
-            },
-        ):
-            reload(self.hass)
-            self.hass.block_till_done()
 
-        assert self.hass.states.get(ENTITY_ID) is None
-        assert not self.hass.services.has_service(script.DOMAIN, "test")
+@pytest.mark.parametrize("value", invalid_configs)
+async def test_setup_with_invalid_configs(hass, value):
+    """Test setup with invalid configs."""
+    assert not await async_setup_component(
+        hass, "script", {"script": value}
+    ), f"Script loaded with wrong config {value}"
+
+    assert 0 == len(hass.states.async_entity_ids("script"))
+
+
+@pytest.mark.parametrize("running", ["no", "same", "different"])
+async def test_reload_service(hass, running):
+    """Verify the reload service."""
+    assert await async_setup_component(
+        hass, "script", {"script": {"test": {"sequence": [{"delay": {"seconds": 5}}]}}}
+    )
+
+    assert hass.states.get(ENTITY_ID) is not None
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5760157)